src/ethereum_spec_tools/evm_tools/t8n/__init__.py

```diff
@@ -442,21 +442,25 @@ def apply_body(self) -> None:
             self.fork.is_after_fork("ethereum.prague")
             and not self.options.state_test
         ):
-            requests_from_execution = (
-                self.fork.process_general_purpose_requests(
-                    deposit_requests,
-                    self.alloc.state,
-                    self.env.block_hashes,
-                    self.env.coinbase,
-                    self.env.block_number,
-                    self.env.base_fee_per_gas,
-                    self.env.block_gas_limit,
-                    self.env.block_timestamp,
-                    self.env.prev_randao,
-                    self.chain_id,
-                    self.env.excess_blob_gas,
+            requests_from_execution = []
+            try:
+                requests_from_execution = (
+                    self.fork.process_general_purpose_requests(
+                        deposit_requests,
+                        self.alloc.state,
+                        self.env.block_hashes,
+                        self.env.coinbase,
+                        self.env.block_number,
+                        self.env.base_fee_per_gas,
+                        self.env.block_gas_limit,
+                        self.env.block_timestamp,
+                        self.env.prev_randao,
+                        self.chain_id,
+                        self.env.excess_blob_gas,
+                    )
                 )
-            )
+            except InvalidBlock as e:
+                self.result.block_exception = f"{e}"
```

Comment:
Should we hoist this up into:

https://github.com/ethereum/execution-specs/blob/8707c283cf21154bd7b3fba2f42f186270d97927/src/ethereum_spec_tools/evm_tools/t8n/__init__.py#L202

?

We can wrap the whole function body in a `try`/`except`, since per-transaction exceptions are caught [here](https://github.com/ethereum/execution-specs/blob/8707c283cf21154bd7b3fba2f42f186270d97927/src/ethereum_spec_tools/evm_tools/t8n/__init__.py#L229).

Reply:
Yes that sounds better. Is `run_blockchain_test` part of `forks/prague`? I cannot see it in this version of the file.

Reply:
I believe so, yeah.